membership-attribute-service/app/actions/AuthAndBackendViaAuthLibAction.scala

```diff
@@ -0,0 +1,30 @@
+package actions
+
+import com.gu.identity.model.User
+import components.{TouchpointBackends, TouchpointComponents}
+import filters.AddGuIdentityHeaders
+import play.api.mvc.{ActionRefiner, Request, Result, Results}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuthAndBackendViaAuthLibAction(touchpointBackends: TouchpointBackends)(implicit ex: ExecutionContext) extends ActionRefiner[Request, AuthenticatedUserAndBackendRequest] {
+  override val executionContext = ex
+
+  override protected def refine[A](request: Request[A]): Future[Either[Result, AuthenticatedUserAndBackendRequest[A]]] = {
+    // On each request via this action, we make a call to IDAPI and see if we can authenticate the user.
+    // The test config and the normal config are the same for IDAPI.
```

Comment:
that is a little bit worrying, if they really are the same, they shouldn't really be in the touchpoint backend. and if they may be different, we should do the test user check before contacting identity if possible